### DISCUSSION TITLE
using nullmodem on a specific port?

### DISCUSSION THREAD

**Discussion by robbiew** [dosemu2/dosemu2#2306] [Q&A]
I'm having trouble getting nullmodem to bind to a specific port... in dosemu.conf, per the documentation:

```
"nullmodem <port_num>" - connect this port to port <port_num> with a
null-modem cable. You need to also set up port <port_num> and connect
it to this port the same way.
 
$_com1 = "nullmodem 5555"
```

dosemu log:
`SER0: none`

I'm trying to use nodejs (e.g. `net.connect({ port: this.nodePort });`) to connect to that port --  but maybe I'm not understanding how dosemu2 handles serial ports for nullmodem? Any tips appreciated!

**Comment by stsp** [comment]
nullmodem is not for that.
You can try "vmodem".
But even that currently only
supports client mode, i.e. it
can only connect. If you need
server mode, for now you need
to use external program, like
socat. In which case in dosemu2
you should use either "exec" or
"pty" modes.

**Comment by jharrison03** [comment]
Their is tty0tty also.  It's a kernel module so less convenient, but I've used this forever with dosemu(2) and tcpser.  It's maintained and tracks current kernels.

https://github.com/freemed/tty0tty

**Reply by stsp** [reply]
How is that any different from socat?
